test/index.js

```diff
@@ -158,12 +224,18 @@ describe('csp middleware', function () {
       })
   })
 
-  it('throws an error reportOnly is true and there is no report-uri', function () {
+  it('throws an error when reportOnly is true and there is no report-uri', function () {
     assert.throws(function () {
       csp({ reportOnly: true })
     }, Error)
   })
 
+  it('does not throw an error when reportOnly is a function and there is no report-uri', function () {
```

Comment:
I feel like this _should_ require a `report-uri`. If it _might_ be `Report-Only`, don't you want a `report-uri`?